app/models/extract_grouping.rb

```diff
@@ -36,16 +36,8 @@ def to_h
       # then flatten each of these groups into a basic list
       Hash[
         extracts_by_classification.group_by{|nest| nest[extractor].data[field]}.map do |subg, ex|
-          [subg, ex.map{|x| x.map{|k,v| pluck_data_key v, field}}.flatten] end
+          [subg, ex.map{|x| x.map{|k,v| v}}.flatten] end
```

Comment:
I find this really hard to read. Can you break up the nested operations to provide some insight into what the code is doing? perhaps..
``` ruby
# sub_groups a  better name? also what does nest mean here? that it's nested..doesn't tell me much about the intention of the code.
extracted_groups = extracts_by_classification.group_by { |nest| nest[extractor].data[field] }
grouped_extract_values = extracted_groups.map do |extracted_group, extracts|
  [extracted_group,  group_values_from_extracts(extracts)]
end
# return the hash conversion here
Hash [ grouped_extract_values ] 

# possible alternative using Hash tap not sure if this is better tbh
{}.tap do |grouped_extract_values
  extracted_groups.each do |extracted_group, extracts|
    grouped_extract_values[extracted_group] =  group_values_from_extracts(extracts)
  end
end

#...extract to method with intention in the name
def group_values_from_extracts(extracts)
  values = extracts.map do |extract| 
    extract.map { |key, value| value }
  end
  values.flatten
end

```

Reply:
Please don't make me try to remember how this works! :P Your point is well taken, I'll see what I can do to tidy this up. It took me most of a day to write it in the first place.

Reply:
i'm not blocking this if you need to get it out - i'll defer to you and the specs here and curse you if i have to fix this code later ;)

Reply:
Nah I'm working on it now. I knew this code was trouble when I wrote it, which is why it had long comments!

Reply:
This actually cleaned up pretty nicely after a little picking-apart.